src/components/molecules/Monaco/codeIntel.ts

```diff
@@ -291,7 +328,70 @@ export async function applyForResource(
     }
   });
 
-  return {newDecorations, newDisposables};
+  return {decorations: newDecorations, disposables: newDisposables};
+}
+
+function applyPolicyIntel(resource: K8sResource): {
+  decorations: monaco.editor.IModelDeltaDecoration[];
+  markers: monaco.editor.IMarkerData[];
+} {
+  const issues = resource.issues?.errors ?? [];
+
+  const glyphs = issues.map(issue => {
+    const rule = issue.rule!;
+    const message = [
+      createMarkdownString(`__${issue.message}:__ ${rule.longDescription.text} ${rule.help.text}`),
+    ].filter(isDefined);
+
+    return createGlyphDecoration(issue.errorPos?.line ?? 1, GlyphDecorationTypes.PolicyIssue, message);
+  });
+
+  const inline: monaco.editor.IModelDeltaDecoration[] = [];
+  // const inline = issues
```

Comment:
This is some work in progress. I want to read up about Monaco in the near future to know better the capabilities and differences. I'll remove this later but wanted it tracked in git.